src/Creator/ProductCombinationCreator.php

```diff
@@ -94,14 +94,19 @@ public function generate(
                     $productNames
                 ));
 
-                $this->commandBus->handle(new GenerateProductCombinationsCommand(
+                $combinationsIds = $this->commandBus->handle(new GenerateProductCombinationsCommand(
                     $newProductId->getValue(),
                     $combinationAttributes,
                 ));
             } else {
                 throw new \RuntimeException(sprintf('Version %s not handled to generate combinations', _PS_VERSION_));
             }
 
+            $combinationsIds = array_map(static function (CombinationId $combination) {
+                return $combination->getValue();
+            }, $combinationsIds);
+
+            $this->associateImages($newProductId->getValue(), $combinationsIds, $numberOfImages);
```

Comment:
Still don't want to add the stock while you're at it? This seems even better than my original suggestion, you can give the array parameter to `associateStockMovements` so it handles the loop internally (default value being `[0]`)

Reply:
(@jolelievre I have a dedicated PR, after this one)